controllers/glanceapi_controller.go

```diff
@@ -683,6 +721,12 @@ func (r *GlanceAPIReconciler) reconcileNormal(ctx context.Context, instance *gla
 	} else if hashChanged {
 		// Hash changed and instance status should be updated (which will be done by main defer func),
 		// so we need to return and reconcile again
+		r.Log.Info(fmt.Sprintf("%s... requeueing", condition.ServiceConfigReadyInitMessage))
+		instance.Status.Conditions.MarkFalse(
+			condition.ServiceConfigReadyCondition,
+			condition.InitReason,
+			condition.SeverityInfo,
+			condition.ServiceConfigReadyInitMessage)
 		return ctrl.Result{}, nil
```

Comment:
This is confusing for me. The log states requeueing but we does not return a requeue result but a success result. Also setting the condition to False and then indicating success is a contradiction to me.

Reply:
right, the wording is probably wrong. I'll remove the `log` message because is a leftover, but the main idea is to mark the condition as False because the next reconcile will update the deployment based on the input that have changed.
Does this makes sense? I think I can't set this condition to True if I still have to reconcile based on the new Input

Reply:
We discussed this over a call. I think the return statement can be removed and let the rest of the reconcile roll out the new hash

Reply:
yeah, this is not required, I can confirm that progressing in the current loop, instead of returning, takes care about rolling new pods out.
I'm going to remove the whole `else if` as part of the next update.